File: Shantanu_Mamgain/Two-Pointer-Leetcode/Interval-List-Intersections.java
class Solution {
    public int[][] intervalIntersection(int[][] firstList, int[][] secondList) {
        int start = 0;
        int end = 0;
        int i = 0;
        int j = 0;
        ArrayList<int[]> list  = new ArrayList <>();
        while(i < firstList.length && j < secondList.length){

            start = Math.max(firstList[i][0], secondList[j][0]);
            end = Math.min(firstList[i][1], secondList[j][1]);
            if(start <= end){
                list.add(new int[] {start , end});
            }
            
            if(firstList[i][1] < secondList[j][1]){
                i++;
            }
            else{
                j++;
            }         
        }

        return list.toArray(new int[list.size()][]);
    }
}

File: Lakshay Garg/Leetcode/Two_Pointers_Leetcode/Interval_List_Intersections.cpp
class Solution {
public:
    vector<vector<int>> intervalIntersection(vector<vector<int>>& f,                vector<vector<int>>& s) {
        vector<vector<int>>v;
        int i=0,j=0;
        while(i<f.size() && j<s.size()) {
            int x=max(f[i][0],s[j][0]);
            int y=min(f[i][1],s[j][1]);
            if(x<=y) v.push_back({x,y});
            if(f[i][1]<s[j][1]) i++;
            else j++;
        }
        return v;
    }
};
